app/components/channel_drawer/channels_list/list/list.js

```diff
@@ -1,157 +1,132 @@
-import React, {Component} from 'react';
+import React, {PureComponent} from 'react';
 import PropTypes from 'prop-types';
 import {
-    FlatList,
+    SectionList,
     Text,
     TouchableHighlight,
     View
 } from 'react-native';
 import {injectIntl, intlShape} from 'react-intl';
 import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
 
-import FormattedText from 'app/components/formatted_text';
-import {wrapWithPreventDoubleTap} from 'app/utils/tap';
-import {changeOpacity} from 'app/utils/theme';
-
 import {General} from 'mattermost-redux/constants';
+import {debounce} from 'mattermost-redux/actions/helpers';
 
 import ChannelItem from 'app/components/channel_drawer/channels_list/channel_item';
 import UnreadIndicator from 'app/components/channel_drawer/channels_list/unread_indicator';
+import {wrapWithPreventDoubleTap} from 'app/utils/tap';
+import {changeOpacity} from 'app/utils/theme';
 
-class List extends Component {
+class List extends PureComponent {
     static propTypes = {
         canCreatePrivateChannels: PropTypes.bool.isRequired,
-        channels: PropTypes.object.isRequired,
-        channelMembers: PropTypes.object,
-        currentChannel: PropTypes.object,
+        directChannelIds: PropTypes.array.isRequired,
+        favoriteChannelIds: PropTypes.array.isRequired,
         intl: intlShape.isRequired,
         navigator: PropTypes.object,
         onSelectChannel: PropTypes.func.isRequired,
+        publicChannelIds: PropTypes.array.isRequired,
+        privateChannelIds: PropTypes.array.isRequired,
         styles: PropTypes.object.isRequired,
-        theme: PropTypes.object.isRequired
-    };
-
-    static defaultProps = {
-        currentChannel: {}
+        theme: PropTypes.object.isRequired,
+        unreadChannelIds: PropTypes.array.isRequired
     };
 
     constructor(props) {
         super(props);
-        this.firstUnreadChannel = null;
+
         this.state = {
-            dataSource: this.buildData(props),
-            showAbove: false
+            sections: this.buildSections(props),
+            showIndicator: false,
+            width: 0
         };
 
         MaterialIcon.getImageSource('close', 20, this.props.theme.sidebarHeaderTextColor).then((source) => {
             this.closeButton = source;
         });
     }
 
-    shouldComponentUpdate(nextProps, nextState) {
-        return !deepEqual(this.props, nextProps, {strict: true}) || !deepEqual(this.state, nextState, {strict: true});
-    }
-
     componentWillReceiveProps(nextProps) {
-        this.setState({
-            dataSource: this.buildData(nextProps)
-        }, () => {
-            if (this.refs.list) {
-                this.refs.list.recordInteraction();
-                this.updateUnreadIndicators({
-                    viewableItems: Array.from(this.refs.list._listRef._viewabilityHelper._viewableItems.values()) //eslint-disable-line
-                });
-            }
-        });
+        if (nextProps !== this.props) {
```

Comment:
I'm pretty sure `nextProps !== this.props` is always true. You could do a shallowEquals on them or check the fields used by buildSections though